core/test/element_util_test.cpp

```diff
@@ -63,3 +64,15 @@ TEST(ElementUtilTest, convertMaskedToZero_accepts_all_types) {
   static_assert(
       std::is_same_v<decltype(convertMaskedToZero(int64_t{}, true)), int64_t>);
 }
+
+TEST(ElementUtilTest, convertToIsoDate_test) {
+  int64_t ts(1595846471200000011);
+  scipp::core::time_point date(ts);
+  std::string unit("ns");
+  EXPECT_EQ(to_iso_date(date, unit), "2020-07-27T10:41:11.200000011\n");
+
+  int64_t ts2(1595846471);
+  scipp::core::time_point date2(ts2);
+  std::string unit2("s");
+  EXPECT_EQ(to_iso_date(date2, unit2), "2020-07-27T10:41:11\n");
+}
```

Comment:
I'm not sure this is the best place for this test, but since `iso_to_date` is a utility, I assumed this was correct.

Reply:
It is not. This file is for the code from `element/util.h`. Since your function is in `string.h` the test should usually go into `string_test.cpp`